src/protocol/vm/state_builder.rs

```diff
@@ -195,7 +195,7 @@ impl VMPoolStateBuilder {
                 self.adapter_contract_path
                     .as_ref()
                     .ok_or_else(|| {
-                        SimulationError::NotInitialized("Adapter contract".to_string())
+                        SimulationError::FatalError("Adapter contract not initialized".to_string())
```

Comment:
We can add more details here - it's specifically has to do with a path error no?